File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/SolutionXML/addsolutionxmlelement.rb
module Asposediagramjava
  module AddSolutionXmlElement
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # initialize SolutionXML object
        solution_xml = Rjb::import('com.aspose.diagram.SolutionXML').new
        
        # set name
        solution_xml.setName("Solution XML")
        
        # set xml value
        solution_xml.setXmlValue("XML Value")
        
        # add SolutionXML element
        diagram.getSolutionXMLs().add(solution_xml)

        # save Visio diagram
        diagram.save(data_dir + "SolutionXmlElement.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Added SolutionXML Element to the Visio Drawing."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithSolutionXMLElements/AddSolutionXmlElement.php
<?php
namespace Aspose\Diagram\WorkingwithSolutionXMLElements;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SolutionXML as SolutionXML;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class AddSolutionXmlElement{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram=new Diagram($dataDir."drawing.vsd");

        # initialize SolutionXML object
        $solution_xml=new SolutionXML();

        # set name
        $solution_xml->setName("Solution XML");

        # set xml value
        $solution_xml->setXmlValue("XML Value");

        # add SolutionXML element
        $diagram->getSolutionXMLs()->add($solution_xml);

        # save Visio diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."SolutionXmlElement.vdx", $saveFileFormat->VDX);
        print "Added SolutionXML Element to the Visio Drawing.".PHP_EOL;
    }
}